server/app/services/applicant/ApplicantService.java

```diff
@@ -1045,4 +1050,118 @@ abstract static class Builder {
+              Optional<Block> block = roApplicantProgramService.getBlock(blockId);
+
+              if (block.isEmpty()) {
+                return CompletableFuture.failedFuture(
+                    new ProgramBlockNotFoundException(programId, blockId));
+              }
+
+              ApplicantQuestion applicantQuestion =
+                  getFirstAddressCorrectionEnabledApplicantQuestion(block.get());
+              AddressQuestion addressQuestion = applicantQuestion.createAddressQuestion();
+
+              Optional<AddressSuggestion> suggestionMaybe =
+                  addressSuggestions.stream()
+                      .filter(
+                          addressSuggestion ->
+                              addressSuggestion.getSingleLineAddress().equals(selectedAddress))
+                      .findFirst();
+
+              ImmutableMap<String, String> questionPathToValueMap =
+                  buildCorrectedAddressAsFormData(
+                      addressQuestion, suggestionMaybe, selectedAddress);
+
+              return CompletableFuture.completedFuture(questionPathToValueMap);
+            });
+  }
+
+  /** Maps address suggestion and corrected state into a form data compatible map */
+  private ImmutableMap<String, String> buildCorrectedAddressAsFormData(
+      AddressQuestion addressQuestion,
+      Optional<AddressSuggestion> suggestionMaybe,
+      String selectedAddress) {
+
+    Map<String, String> questionPathToValueMap = new HashMap<>(Map.of());
+
+    if (suggestionMaybe.isPresent()) {
+      AddressSuggestion suggestion = suggestionMaybe.get();
+      Address address = suggestion.getAddress();
+      AddressLocation location = suggestion.getLocation();
+
+      questionPathToValueMap.put(addressQuestion.getStreetPath().toString(), address.getStreet());
+      questionPathToValueMap.put(addressQuestion.getLine2Path().toString(), address.getLine2());
+      questionPathToValueMap.put(addressQuestion.getCityPath().toString(), address.getCity());
+      questionPathToValueMap.put(addressQuestion.getStatePath().toString(), address.getState());
+      questionPathToValueMap.put(addressQuestion.getZipPath().toString(), address.getZip());
+      questionPathToValueMap.put(
+          addressQuestion.getLatitudePath().toString(), location.getLatitude().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getLongitudePath().toString(), location.getLongitude().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getWellKnownIdPath().toString(), location.getWellKnownId().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getCorrectedPath().toString(),
+          CorrectedAddressState.CORRECTED.toString());
+    } else if (selectedAddress.equals(AddressCorrectionBlockView.USER_KEEPING_ADDRESS_VALUE)) {
+      questionPathToValueMap.put(
+          addressQuestion.getCorrectedPath().toString(),
+          CorrectedAddressState.AS_ENTERED_BY_USER.toString());
```

Comment:
**question (blocking):** I thought we were requiring applicants to use corrected addresses so that service area verification can work? @seanmalbert ?

Reply:
Yes, if service area validation is enabled and specified for a block, then address correction should not allow the "as entered" flow.

Reply:
It's being allowed in conditionally. It's not an option if the service area validation is enabled as Sean said, but it'll be an option if correction is enable, but not the validation part. It allows the program more flexibility in their intent on a question.

Reply:
Got it, please add that to the javadoc in the public method

Reply:
done